peerstore/src/peer_info.rs

```diff
@@ -108,64 +107,30 @@ pub enum AddAddrBehaviour {
     IgnoreTtlIfInferior,
 }
 
-impl Serialize for PeerInfo {
+/// Same as `Multiaddr`, but serializable.
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct SerdeMultiaddr(Multiaddr);
```

Comment:
Why don't we add `serde::{Serialize, Deserialize}` impls directly to `Multiaddr`? We could utilise [`Serializer::is_human_readable`](https://docs.serde.rs/serde/ser/trait.Serializer.html#method.is_human_readable) to support binary and text (with text being used for JSON).